README.md

```diff
@@ -2,115 +2,91 @@
+             	--build-arg GEN_PYTORCH_EXT=1 \
+             	--build-arg GEN_OPENGL_EXT=1 \
+             	.docker run -it --rm --gpus=all vpf-gpu-all``
+
+  If using Docker via [Nvidia Container Runtime](https://developer.nvidia.com/nvidia-container-runtime),
+please make sure to enable the `video` driver capability: https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/user-guide.html#driver-capabilities via the `NVIDIA_DRIVER_CAPABILITIES` environment variable in the container or the `--gpus` command line parameter (e.g.
 `docker run -it --rm --gpus 'all,"capabilities=compute,utility,video"' nvidia/cuda:12.1.0-base-ubuntu22.04`).
 
-Please note that some examples have additional dependencies https://github.com/NVIDIA/VideoProcessingFramework/blob/73a14683a17c8f1c7fa6dd73952f8813bd34a11f/setup.py#L26-L31
-that need to be installed via pip. Samples using PyTorch will require an optional extension which can be installed via
-```bash
-pip install src/PytorchNvCodec  # install Torch extension if needed (optional), requires "torch" to be installed before
+- ### building and installing VPF
+  VPF provides python bindings over C++ API’s on top of Video Codec SDK.  Compiling VPF source code builds VPF binary (WHL file of PyNvCodec python module). There are two ways to build VPF binaries on client machines.
+  - “pip install” that automatically downloads source code and builds VPF binaries on client machines. This is quick and is the preferred way. 
+  - Alternatively you can download VPF source code and build it from source by following the instructions provided. 
+
+## Installation using pip command
+- ### Installing using pip command on Windows
+  - Indicate path to your FFMPEG installation (with subfolders `bin` with DLLs, `include`, `lib`) in “TC_FFMPEG_ROOT” environment variable
+  - Run the following command.
+    ``set SKBUILD_CONFIGURE_OPTIONS="-DTC_FFMPEG_ROOT=C:/path/to/your/ffmpeg/installation/ffmpeg/"``
+  - Run the following command.
+    - “pip3 install git+https://github.com/NVIDIA/VideoProcessingFramework” or
+    - If you have already cloned this repository just run “pip3 install .”
+
+- ### Installing using pip command on Ubuntu 20.04
+  1. Please follow the instructions below to install ffmpeg
+    (https://docs.nvidia.com/video-technologies/video-codec-sdk/12.0/ffmpeg-with-nvidia-gpu/index.html#compiling-for-linux)
+  2. Install dependencies ( String XXX here represents driver version, replace XXX in libnvidia-encode-XXX, libnvidia-decode-XXX with your driver version, for e.g.  if you had installed r525 driver, replace XXX with 525)
+    - libnvidia-encode is part of the driver meta package so if your driver works fine you might already have it.
+  3. Run the command below to install and update dependencies
+    ``apt install -y  libavfilter-dev \ libavformat-dev \ libavcodec-dev \ libswresample-dev \ libavutil-dev\ wget \ 	cmake \ build-essential \    	git \      	libnvidia-encode-XXX \ libnvidia-decode-XXX''
```

Comment:
can you fix the formatting?

I would prefer to just have a single terminal block that sets up everything instead of text. Python packages are expected to be installed via single command. We should at least make this as simple as possible